providers/hostingde/hostingdeProvider.go

```diff
@@ -159,7 +159,7 @@ func (hp *hostingdeProvider) GetDomainCorrections(dc *models.DomainConfig) ([]*m
 	var DnsSecOptions *dnsSecOptions = nil
 
 	// ensure that publishKsk is set for domains with AutoDNSSec
-	if existingAutoDNSSecEnabled == desiredAutoDNSSecEnabled == true {
+	if existingAutoDNSSecEnabled == desiredAutoDNSSecEnabled && desiredAutoDNSSecEnabled == true {
```

Comment:
If desiredAutoDNSSecEnabled is a bool, you don't need to compare it to true.

Isn't this equivalent?

```
if desiredAutoDNSSecEnabled && existingAutoDNSSecEnabled {
```

Reply:
Absolutely right. That's better.